scripts/deleteExpiredAccessTokens.js

```diff
@@ -0,0 +1,178 @@
+    fluid = require("infusion");
+
+fluid.require("%gpii-universal/scripts/shared/dbRequestUtils.js");
+
+var gpii = fluid.registerNamespace("gpii");
+fluid.registerNamespace("gpii.accessTokens");
+
+fluid.setLogging(fluid.logLevel.INFO);
+
+// Handle command line
+if (process.argv.length < 3) {
+    fluid.log("Usage: node deleteExpiredAccessTokens.js $COUCHDB_URL [--deleteAll]");
+    process.exit(1);
+}
+
+/**
+ * Create a set of options for this script.
+ * The options are based on the command line parameters and a set of database
+ * constants.
+ * @param {Array} processArgv - The command line arguments.
+ * @return {Object} - The options.
+ */
+gpii.accessTokens.initOptions = function (processArgv) {
+    var options = {};
+    options.couchDbUrl = processArgv[2];
+
+    // Ignore time of expiration and delete all access tokens?
+    options.deleteAll = fluid.contains(processArgv, "--deleteAll");
+
+    // Set up database specific options
+    options.accessTokensUrl = options.couchDbUrl + "/_design/views/_view/findAuthorizationByAccessToken";
+    options.accessTokens = [];
+    options.parsedCouchDbUrl = url.parse(options.couchDbUrl);
+    options.postOptions = {
+        hostname: options.parsedCouchDbUrl.hostname,
+        port: options.parsedCouchDbUrl.port,
+        path: "/gpii/_bulk_docs",
+        auth: options.parsedCouchDbUrl.auth,
+        method: "POST",
+        headers: {
+            "Accept": "application/json",
+            "Content-Length": 0,          // IMPORTANT: FILL IN PER REQUEST
+            "Content-Type": "application/json"
+        }
+    };
+    fluid.log("COUCHDB_URL: '" +
+        options.parsedCouchDbUrl.protocol + "//" +
+        options.parsedCouchDbUrl.hostname + ":" +
+        options.parsedCouchDbUrl.port +
+        options.parsedCouchDbUrl.pathname + "'"
+    );
+    return options;
+};
+
+/**
+ * Create the step that retrieves access tokens from the database:  either
+ * only the expired tokens, or all of them.
+ * @param {Object} options - Access tokens URL and whether to filter:
+ * @param {Array} options.accessTokensUrl - The url for retrieving all of the
+ *                                          access tokens in the database.
+ * @param {Boolean} options.deleteAll - Flag indicating whether to ignore the
+ *                                      expiration date of the access tokens.
+ * @return {Promise} - A promise that resolves retrieving the tokens.
+ */
+gpii.accessTokens.retrieveExpiredAccessTokens = function (options) {
+    var details = {
+        requestUrl: options.accessTokensUrl,
+        requestErrMsg: "Error retrieving access tokens from the database: ",
+        responseDataHandler: gpii.accessTokens.filterExpiredAccessTokens,
+        responseErrMsg: "Error retrieving access tokens from database: "
+    };
+    return gpii.dbRequest.configureStep(details, options);
+};
+
+/**
+ * Given all the access tokens from the database, filter out only the ones that
```

Comment:
Please improve this comment to include that this function filters out all access tokens when deleteAll flag is set to true.

Reply:
Oh, this comment already says "The filter is not applied.. ". Probably rephrase that line to indicate the use of options.deleteAll. Thanks.